libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -968,6 +1009,15 @@ void test_MQTT_SerializePublish( void )
                                     remainingLength,
                                     &fixedBuffer );
     TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    memset( ( void * ) expectedPacket, 0x00, sizeof( expectedPacket ) );
+    pIterator = expectedPacket;
+    /* Dup = 0x8, QoS2 = 0x4, Retain = 0x1. 8 + 4 + 1 = 0xD. */
+    *pIterator++ = MQTT_PACKET_TYPE_PUBLISH | 0xD;
+    pIterator += encodeRemainingLength( pIterator, remainingLength );
+    pIterator += encodeString( pIterator, publishInfo.pTopicName, publishInfo.topicNameLength );
+    *pIterator++ = UINT16_HIGH_BYTE( PACKET_ID );
+    *pIterator++ = UINT16_LOW_BYTE( PACKET_ID );
+    TEST_ASSERT_EQUAL_MEMORY( expectedPacket, buffer, packetSize );
```

Comment:
I think it would add value if we can add the buffer validation as part of all the Success cases. Line 976. What are your thoughts?

Reply:
I'm not sure I understand. Nothing is written into the buffer by line 976, that's just making sure the calculated packet size does not exceed the buffer size

Reply:
Sorry, wrong line number, I meant 982

Reply:
That buffer is verified in line 993. Everything between 982 and 993 just generates the expected packet. Should it be rearranged so the buffer verification is closer to when the status is tested?

Reply:
Yea. Ok. I see the problem. I missed to notice the assert in 993 and was thinking that it is setting the expected buffer for the next test. I think it is ok as it is.